vizro-core/tests/unit/vizro/actions/_callback_mapping/test_get_action_callback_mapping.py

```diff
@@ -0,0 +1,267 @@
+            argument="outputs",
+        )
+        assert result == action_callback_outputs_expected
+
+    @pytest.mark.parametrize(
+        "export_data_action_targets, export_data_outputs_expected",
+        [
+            (None, ["scatter_chart", "scatter_chart_2"]),
+            (["scatter_chart"], ["scatter_chart"]),
+            (["scatter_chart", "scatter_chart_2"], ["scatter_chart", "scatter_chart_2"]),
+        ],
+        indirect=True,
+    )
+    def test_export_data_mapping_outputs(self, export_data_action_targets, export_data_outputs_expected):
+        export_data_action = model_manager["export_data_action"]
+        export_data_action.function = (
+            export_data(targets=export_data_action_targets) if export_data_action_targets else export_data()
+        )
+
+        result = _get_action_callback_mapping(
+            action_id="export_data_action",
+            argument="outputs",
+        )
+
+        assert result == export_data_outputs_expected
+
+    @pytest.mark.parametrize(
+        "export_data_action_targets, export_data_components_expected",
+        [
+            (None, ["scatter_chart", "scatter_chart_2"]),
+            (["scatter_chart"], ["scatter_chart"]),
+            (["scatter_chart", "scatter_chart_2"], ["scatter_chart", "scatter_chart_2"]),
+        ],
+        indirect=True,
+    )
+    def test_export_data_mapping_components(self, export_data_action_targets, export_data_components_expected):
+        export_data_action = model_manager["export_data_action"]
+        export_data_action.function = (
+            export_data(targets=export_data_action_targets) if export_data_action_targets else export_data()
+        )
+
+        result = _get_action_callback_mapping(
+            action_id="export_data_action",
+            argument="components",
+        )
+        assert repr(result) == repr(export_data_components_expected)
+
+    def test_known_action_unknown_argument(self):
+        result = _get_action_callback_mapping(
+            action_id="export_data_action",
+            argument="unknown-argument",
+        )
+        assert result == {}
+
+    @pytest.mark.parametrize(
+        "argument, expected",
+        [
+            ("inputs", {}),
+            ("outputs", {}),
+            pytest.param("components", [], marks=pytest.mark.xfail()),
+            ("unknown-argument", {}),
+        ],
+    )
+    def test_custom_action_mapping(self, argument, expected):
+        result = _get_action_callback_mapping(
+            action_id="custom_action",
+            argument=argument,
+        )
+        assert result == expected
+
+    @pytest.mark.parametrize(
+        "argument, expected",
+        [
+            ("inputs", {}),
+            ("outputs", {}),
+            pytest.param("components", [], marks=pytest.mark.xfail()),
+            ("unknown-argument", {}),
+        ],
+    )
+    def test_custom_action_with_known_name_mapping(self, argument, expected):
```

Comment:
Hm what is this testing? I feel its clear if you give it a different ID, then it will not find it in the lookup right?

Reply:
It tests fetching inputs/outputs/components/unknown-argument mapping for the custom action which function has the name as some predefined action. 

To be honest I don't understand what do you mean? Do I need just to rename this test to something like `test_custom_action_with_predefined_name_mapping`, or also to define this action inside the test function?

Reply:
Hm I now cannot fully recall myself. Just had a look and just have one small other suggestion